src/acl.ts

```diff
@@ -126,97 +126,97 @@ function getContainerPath(resourcePath: string): string {
 }
 
 /**
- * Verify whether an ACL was found for the given LitDataset.
+ * Verify whether an ACL was found for the given Resource.
  *
- * A LitDataset fetched with [[unstable_fetchLitDatasetWithAcl]] _might_ have an ACL attached, but
+ * A Resource fetched with its ACL (e.g. using [[unstable_fetchLitDatasetWithAcl]]) _might_ have an resource ACL attached, but
  * we cannot be sure: it might be that none exists for this specific Resource (in which case the
  * fallback ACL applies), or the currently authenticated user (if any) might not have Control access
  * to the fetched Resource.
  *
- * This function verifies that the LitDataset's ACL is accessible.
+ * This function verifies that the Resource's ACL is accessible.
  *
- * @param dataset A [[LitDataset]] that might have an ACL attached.
+ * @param acl A Resource that might have an ACL attached.
  * @returns Whether `dataset` has an ACL attached.
  */
-export function unstable_hasResourceAcl<Dataset extends unstable_Acl>(
-  dataset: Dataset
-): dataset is Dataset & {
-  acl: { resourceAcl: Exclude<unstable_Acl["acl"]["resourceAcl"], undefined> };
+export function unstable_hasResourceAcl<Acl extends unstable_WithAcl["acl"]>(
```

Comment:
It kinda undermines your entire PR (sorry!), but... I think I'd feel more comfortable keeping this function as accepting the full Resource, i.e.:

```suggestion
export function unstable_hasResourceAcl<ResourceWithAcl extends unstable_WithAcl>(
```

The reason for this is that generally, I _think_ we don't want to require developers to first extract the ACL file from the Resource, and only _then_ be able to work with it. I'm afraid that that will lead to situations where developers are going to be required to pass _both_ around all over the place.

I'm not 100% sure about this - it might very well be that, once we get to writing ACLs, there's no avoiding having the developer manually set it. But I'd prefer to postpone this refactor until we really can no longer avoid it, and until then try to maintain an API where the external-facing methods just operate on the Resource itself.

I suppose I should have added an E2E test for just the reading of ACLs already... I'll make sure I'll do that in for the "writing" PR.

Reply:
I see your point. I was thinking about this change feature-wise, but design-wise, I think you are right. My motivation was that when you only fetch a resource's information and ACL, you don't have the resource itself, but rather either the resource info or the ACL, and I was concerned that these two being separate pieces of data, it would be burdensome to not take one of these as a parameter, instead of a Resource-like object with an `acl` or `resourceInfo` member. 

But I think that this could not be an issue if we design the functions to only fetch metadata accordingly. I'm not sure 
```typescript
async function fetchResourceInfo(
  url: UrlString,
  options: Partial<typeof defaultFetchOptions> = defaultFetchOptions
): Promise<WithResourceInfo["resourceInfo"]>
```
should change to return `Promise<WithResourceInfo>`, because `const myInfo = (await fetchResourceInfo("someUrl")).resourceInfo` doesn't look right (IMHO), but I wouldn't mind having 
```typescript
async function fetchResourceInfoAndAcl(
  url: UrlString,
  options: Partial<typeof defaultFetchOptions> = defaultFetchOptions
): Promise<WithResourceInfo & unstable_WithAcl>
```
With the latter, the fetched object can be passed to `unstable_hasResourceAcl`, so it still holds to the motivation for this PR, which is to make it easy to work without fetching the body of the resource

Reply:
> `const myInfo = (await fetchResourceInfo("someUrl")).resourceInfo` doesn't look right (IMHO)

Agreed, so hopefully by keeping that an internal method we can sidestep this issue :P

(I also don't mind whether it's `fetchResourceInfoAndAcl` or `fetchResourceInfoWithAcl`, but make sure that it's consistent with `fetchLitDatasetWithAcl`, i.e. rename that if necessary.)